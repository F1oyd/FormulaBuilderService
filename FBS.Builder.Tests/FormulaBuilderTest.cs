using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using NUnit.Framework;

namespace FBS.Builder.Tests
{
    [TestFixture]
    public class FormulaBuilderTest
    {
        [Test]
        [UseReporter(typeof(VisualStudioReporter))]
        public void TestBuilder()
        {
            // Arrange
            var input = @"<request><expression><operation>plus</operation><operand><const>20</const></operand><operand>
                        <expression><operation>minus</operation><operand><const>10</const></operand><operand><const>5</const></operand>
                        </expression></operand></expression></request>";
            // Act
            var builder = new FormulaBuilder(input);
            var result = builder.Build();

            // Assert
            Approvals.Verify(result);
        }

        [Test]
        [UseReporter(typeof(VisualStudioReporter))]
        public void TestOperationValidation()
        {
            // Arrange
            var input = @"<request><expression><operation>plus</operation><operand><const>20</const></operand><operand>
                        <expression><operation>bla</operation><operand><const>10</const></operand><operand><const>5</const></operand>
                        </expression></operand></expression></request>";
            // Act
            var builder = new FormulaBuilder(input);
            var result = builder.Build();

            // Assert
            Approvals.Verify(result);
        }


        [Test]
        [UseReporter(typeof(VisualStudioReporter))]
        public void TestExpression()
        {
            // Arrange
            var input = @"<request><expression><operation>mul</operation><operand><const>20</const></operand><operand>
                        <expression><operation>plus</operation><operand><const>10</const></operand><operand><expression>
                        <operation>div</operation><operand><const>100</const></operand><operand><const>50</const></operand>
                        </expression></operand></expression></operand></expression></request>";
            // Act
            var builder = new FormulaBuilder(input);
            var result = builder.Build();

            // Assert
            Approvals.Verify(result);
        }
    }
}
