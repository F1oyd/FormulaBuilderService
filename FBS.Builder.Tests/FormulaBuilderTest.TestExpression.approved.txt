<response>
  <result>20 * (10 + (100 / 50))</result>
  <errors>
  </errors>
</response>